numba/datamodel/packer.py

```diff
@@ -2,25 +2,61 @@
 
 from collections import deque
 
-from numba import types
+from numba import cgutils, types
 
 
-class _OmittedArgDataModel(object):
+
+class DataPacker(object):
     """
-    A fake data model for omitted arguments.
-    Only used in this module.
+    A helper to pack a number of typed arguments into a data structure.
+    Omitted arguments (i.e. values with the type `Omitted`) are automatically
+    skipped.
     """
-    # Omitted arguments don't produce any LLVM function argument
+    # XXX should DataPacker be a model for a dedicated type?
```

Comment:
After trying this out a bit, it seems making this a data model wouldn't make things any cleaner (since the input to the data model isn't a regular LLVM value, but a tuple containing both LLVM values or empty tuples (for omitted arguments)). I think keeping the separate `DataPacker` class at least stresses that something special is happening here...